client/src/rubin/nublado/client/_util.py

```diff
@@ -61,35 +69,101 @@ def normalize_source_by_cell(notebook: str) -> list[str]:
        A list of all non-empty source lines in a cell as a single Python
     string.  Each cell's source lines (with newline as the line separator) will
     be a separate item of the returned list.
+
+    Notes
+    -----
+    This is what the contents API returns, although the text of the notebook
+    on disk will have each source line as its own entry within a list of
+    strings.  So we will convert it to API form first and then return the
+    source item from each cell.
     """
+    notebook = notebook_to_api_form(notebook)
+    obj = json.loads(notebook)
     return [
-        "\n".join(
-            [
-                x.rstrip("\n")
-                for y in extract_source_by_cell(notebook)
-                for x in y
-                if x.rstrip("\n")
-            ]
-        )
+        x["source"]
+        for x in obj["cells"]
+        if x["cell_type"] == "code" and "source" in x and x["source"]
     ]
 
 
-def extract_source_by_cell(notebook: str) -> list[list[str]]:
-    """Extract all non-empty "code" cells' "source" lines as a list of strings.
+def source_list_by_cell(notebook: str) -> list[list[str]]:
```

Comment:
Would it make sense to merge `source_list_by_cell` & `source_string_by_cell` into one method, which will take the `NotebookForm` as a parameter and call out the according method (notebook_to_x.. ) ?

Reply:
Oh, I should just get rid of NotebookForm.  Since I'm using the file API instead of the contents one I don't need it.